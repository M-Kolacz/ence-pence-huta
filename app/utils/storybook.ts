import { type ComponentType, type ComponentProps } from 'react'
import { type MY_VIEWPORTS, allModes } from '../../.storybook/modes.ts'

export const disableControls = <Component extends ComponentType<any>>(
	controls: (keyof ComponentProps<Component>)[],
) => {
	const disabledOptions = {}

	controls.forEach(control =>
		Object.assign(disabledOptions, {
			[control]: {
				table: {
					disable: true,
				},
			},
		}),
	)

	return disabledOptions
}

export const setViewport = (device: keyof typeof MY_VIEWPORTS) => ({
	viewport: {
		defaultViewport: device,
	},
	chromatic: {
		modes: {
			[device]: allModes[device],
		},
	},
})
