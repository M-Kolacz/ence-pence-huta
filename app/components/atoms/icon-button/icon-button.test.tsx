import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import { getStoriesTestCases } from '#app/utils/storybook.ts'
import * as IconButtonStories from './icon-button.stories.tsx'

const testCases = getStoriesTestCases(IconButtonStories)

describe('IconButton component', () => {
	test.each(testCases)(
		'snapshot test of %s story',
		async (_storyName, Story) => {
			if (!Story) throw new Error('Missing story to test!')

			const { container } = render(<Story />)
			expect(container).toMatchSnapshot()
		},
	)

	test.each(testCases)('a11y test of %s story', async (_storyName, Story) => {
		if (!Story) throw new Error('Missing story to test!')

		const { container } = render(<Story />)

		const result = await axe(container)

		expect(result).toHaveNoViolations()
	})
})
