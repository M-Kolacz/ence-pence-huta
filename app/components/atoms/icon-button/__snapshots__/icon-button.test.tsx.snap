// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconButton component > snapshot test of Component story 1`] = `
<div>
  <button
    class="inline-flex h-7 w-7 items-center justify-center whitespace-nowrap rounded-md border border-base bg-neutral-0 p-2 transition-colors hover:bg-strong focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary-600 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50"
    title="Icon Button"
  >
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#cereal"
      />
    </svg>
  </button>
</div>
`;
